app-modules/inventory-management/src/Filament/Resources/MaintenanceProviderResource/Pages/ListMaintenanceProviders.php

```diff
@@ -0,0 +1,110 @@
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AidingApp\InventoryManagement\Filament\Resources\MaintenanceProviderResource\Pages;
+
+use AidingApp\InventoryManagement\Filament\Resources\MaintenanceProviderResource;
+use AidingApp\InventoryManagement\Models\MaintenanceProvider;
+use Filament\Actions\CreateAction;
+use Filament\Notifications\Notification;
+use Filament\Resources\Pages\ListRecords;
+use Filament\Tables\Actions\BulkActionGroup;
+use Filament\Tables\Actions\DeleteAction;
+use Filament\Tables\Actions\DeleteBulkAction;
+use Filament\Tables\Actions\EditAction;
+use Filament\Tables\Columns\TextColumn;
+use Filament\Tables\Table;
+use Illuminate\Database\Eloquent\Collection;
+
+class ListMaintenanceProviders extends ListRecords
+{
+    protected static string $resource = MaintenanceProviderResource::class;
+
+    public function table(Table $table): Table
+    {
+        return $table
+            ->columns([
+                TextColumn::make('name')
+                    ->searchable()
+                    ->sortable(),
+            ])
+            ->actions([
+                EditAction::make(),
+                DeleteAction::make()
+                    ->hidden(fn ($record) => $record->assets_count > 0),
+            ])
+            ->bulkActions([
+                BulkActionGroup::make([
+                    DeleteBulkAction::make()
+                        ->action(function (Collection $records) {
+                            $successfullyDeleted = 0;
+                            $unsuccessfullyDeleted = 0;
+
+                            foreach ($records as $record) {
+                                /** @var MaintenanceProvider $record */
+                                if ($record->assets_count > 0) {
```

Comment:
This will need to change once you get rid of the column and get this through the relationship.